src/components/InputTypes/Code.tsx

```diff
@@ -1,11 +1,37 @@
 /** @jsx jsx */
 import { jsx } from "@emotion/react";
-import React, { ChangeEvent } from "react";
+import React, { useEffect } from "react";
+
+import { basicSetup } from "@codemirror/basic-setup";
+import { python } from "@codemirror/lang-python";
+import { EditorState } from "@codemirror/state";
+import { oneDark } from "@codemirror/theme-one-dark";
+import { EditorView, ViewUpdate } from "@codemirror/view";
 
 interface CodeProps {
-    handler: (event: ChangeEvent<HTMLInputElement>) => void
+    handler: (newContent: string) => void,
+    questionId: string,
 }
 
 export default function Code(props: CodeProps): JSX.Element {
-    return <input type="text" className="text" onChange={props.handler}/>;
+    const onUpdate = () => EditorView.updateListener.of((v: ViewUpdate) => {
+        props.handler(v.state.doc.toString());
```

Comment:
We should place an upper limit on the length of content that is rendered with the theme. If the content passes that limit, don't do rendering on it to prevent lag.

Reply:
The default is 10,000 already which seems to work well, but I'll explicitly define it.